api/src/pools/subscription-pools/index.ts

```diff
@@ -2,35 +2,48 @@ import streams, { Author, Subscriber } from '../../streams-lib/wasm-node/iota_st
 import * as SubscriptionDb from '../../database/subscription';
 import { SubscriptionType } from '../../models/types/subscription';
 import { toBytes } from '../../utils/text';
+import { subSeconds } from 'date-fns';
 
-// TODO#39 use more robust object pool: https://github.com/electricessence/TypeScript.NET/blob/master/source/System/Disposable/ObjectPool.ts
 export class SubscriptionPool {
-	private static instance: SubscriptionPool;
-	private readonly node: string;
-	private authors: { identityId: string; channelAddress: string; author: Author }[];
-	private subscribers: { identityId: string; channelAddress: string; subscriber: Subscriber }[];
+	private secondsToLive = 20;
+	private interval: NodeJS.Timeout;
+	authors: { identityId: string; channelAddress: string; author: Author; created: Date }[];
+	subscribers: { identityId: string; channelAddress: string; subscriber: Subscriber; created: Date }[];
 
-	private constructor(node: string) {
+	constructor(private readonly node: string, private readonly maxPoolSize = 65000) {
 		this.authors = [];
 		this.subscribers = [];
-		this.node = node;
 	}
 
-	public static getInstance(node: string): SubscriptionPool {
-		if (!SubscriptionPool.instance) {
-			SubscriptionPool.instance = new SubscriptionPool(node);
-		}
-		return SubscriptionPool.instance;
+	startInterval() {
+		this.interval = setInterval(() => this.clearObsoleteObjects(), this.secondsToLive * 1000);
+	}
+
+	stopInterval() {
+		clearInterval(this.interval);
 	}
 
 	add(channelAddress: string, subscription: Author | Subscriber, identityId: string, isAuthor: boolean) {
 		if (isAuthor) {
-			this.authors = [...this.authors, { author: <Author>subscription, channelAddress, identityId }];
+			if (this.authors.length === this.maxPoolSize) {
+				this.authors.pop();
+			}
+			const newAuthor = { author: <Author>subscription, channelAddress, identityId, created: new Date(Date.now()) };
```

Comment:
`new Date(Date.now())`  is equal to `new Date()`

Reply:
yes but I need to mock the date function in a test but I am only able to mock Date.now that why it is written like this here